src/trading/Orders.sol

```diff
@@ -161,48 +152,46 @@ contract Orders is DelegationTarget, IOrders {
     }
 
     // FIXME: Remove first and last parameters, as they are no longer used
-    function saveOrder(bytes32, Trading.TradeTypes _type, IMarket _market, uint256 _fxpAmount, int256 _fxpPrice, address _sender, uint8 _outcome, uint256 _fxpMoneyEscrowed, uint256 _fxpSharesEscrowed, bytes32 _betterOrderId, bytes32 _worseOrderId, uint256 _tradeGroupId, uint256) public onlyWhitelistedCallers returns (bytes32 _orderId) {
-        require(_type == Trading.TradeTypes.Bid || _type == Trading.TradeTypes.Ask);
+    function saveOrder(bytes32, Order.TradeTypes _type, IMarket _market, uint256 _fxpAmount, int256 _fxpPrice, address _sender, uint8 _outcome, uint256 _fxpMoneyEscrowed, uint256 _fxpSharesEscrowed, bytes32 _betterOrderId, bytes32 _worseOrderId, uint256 _tradeGroupId, uint256) public onlyWhitelistedCallers returns (bytes32 _orderId) {
         require(_outcome < _market.getNumberOfOutcomes());
         _orderId = getOrderId(_type, _market, _fxpAmount, _fxpPrice, _sender, block.number, _outcome, _fxpMoneyEscrowed, _fxpSharesEscrowed);
         insertOrderIntoList(_orderId, _type, _market, _outcome, _fxpPrice, _betterOrderId, _worseOrderId);
         orders[_orderId].fxpPrice = _fxpPrice;
```

Comment:
The new `Order` object has additional fields that are not populated here.  We should make sure to fully populate the object (e.g., `orderId`) so someone can't write code that accesses uninitialized fields on the `Order`.  I think there may be a couple more besides `orderId`?

Reply:
Yeah, I can add:
```
        orders[_orderId].orders = this;
        orders[_orderId].market = _market;
        orders[_orderId].id = _orderId;
        orders[_orderId].betterOrderId = _betterOrderId;
        orders[_orderId].worseOrderId = _worseOrderId;
```
(I'll leave out tradeType and outcome, since you said they are already in your branch.)

Reply:
Don't we need to add lines to removeOrder() for these values as well?

Reply:
Yeah, probably.  We want removeOrder to clear out everything.  Though, maybe `delete orders[_orderId]` is enough?  Would need to test to see if you get a full gas refund for that operation.

Reply:
Ok, should I add a TODO for testing `delete orders[_orderId]`, or do you think this should be done before merging?

Reply:
Either update `removeOrder` to clear out all members and add a TODO to investigate `delete orders[_orderId]` or test `delete orders[_orderId]` and see if it refunds gas.  Up to you which one you want to do.  😄

Reply:
I'll add a TODO to look at this next week so we can get this merged a bit sooner.